pkg/controller/provider/validation.go

```diff
@@ -400,3 +412,429 @@ func isValidNFSPath(nfsPath string) bool {
+func (r *Reconciler) getHostsForSSHValidation(provider *api.Provider) []hostInfo {
+	sdkEndpoint := provider.Spec.Settings[api.SDK]
+	isDirectESXi := (sdkEndpoint == api.ESXI)
+
+	r.Log.Info("SSH validation: provider connection type",
+		"provider", provider.Name,
+		"sdkEndpoint", sdkEndpoint,
+		"isDirectESXi", isDirectESXi)
+
+	if isDirectESXi {
+		// For direct ESXi connections, extract IP from provider URL
+		r.Log.Info("SSH validation: direct ESXi connection detected, extracting IP from provider URL",
+			"provider", provider.Name,
+			"providerURL", provider.Spec.URL)
+
+		providerURL, err := url.Parse(provider.Spec.URL)
+		if err != nil {
+			provider.Status.SetCondition(libcnd.Condition{
+				Type:     SSHReadiness,
+				Status:   False,
+				Reason:   "InvalidProviderURL",
+				Category: Warn,
+				Message:  fmt.Sprintf("Cannot validate SSH readiness: failed to parse provider URL: %v", err),
+			})
+			return nil
+		}
+
+		// Extract hostname/IP from URL (could be hostname:port or just hostname)
+		hostIP := providerURL.Hostname()
+		if hostIP == "" {
+			provider.Status.SetCondition(libcnd.Condition{
+				Type:     SSHReadiness,
+				Status:   False,
+				Reason:   "NoHostIPInURL",
+				Category: Warn,
+				Message:  fmt.Sprintf("Cannot validate SSH readiness: no host/IP found in provider URL: %s", provider.Spec.URL),
+			})
+			return nil
+		}
+
+		r.Log.Info("SSH validation: extracted IP from provider URL",
+			"provider", provider.Name,
+			"hostIP", hostIP)
+
+		// Single host for direct ESXi
+		return []hostInfo{{name: "ESXi", ip: hostIP}}
+	}
+
+	// For vCenter connections, use inventory to get ESXi hosts
+	r.Log.Info("SSH validation: vCenter connection, attempting to get collector", "provider", provider.Name, "namespace", provider.Namespace)
+	collector, found := r.container.Get(provider)
+	if !found {
+		r.Log.Error(nil, "SSH validation: collector not found for provider - SSH readiness check failed", "provider", provider.Name, "namespace", provider.Namespace)
+		provider.Status.SetCondition(libcnd.Condition{
+			Type:     SSHReadiness,
+			Status:   False,
+			Reason:   "InventoryCollectorNotFound",
+			Category: Warn,
+			Message:  "Cannot validate SSH readiness: inventory collector not found. Ensure the provider is properly configured and inventory collection has started.",
+		})
+		return nil
+	}
+
+	r.Log.Info("SSH validation: collector found, checking parity", "provider", provider.Name, "hasParity", collector.HasParity())
+	if !collector.HasParity() {
+		r.Log.Error(nil, "SSH validation: collector does not have parity yet - SSH readiness check failed", "provider", provider.Name)
+		provider.Status.SetCondition(libcnd.Condition{
+			Type:     SSHReadiness,
+			Status:   False,
+			Reason:   "InventoryNotReady",
+			Category: Warn,
+			Message:  "Cannot validate SSH readiness: inventory collection has not completed (parity not reached). Wait for inventory to finish loading.",
+		})
+		return nil
+	}
```

Comment:
This is checked in previous validations, we should not start validating this if the DB is not ready